docs/proposals/08-shoot-apiserver-via-sni.md

```diff
@@ -206,136 +206,168 @@ Another issue occurs when the client tries to talk to the apiserver via the in-c
-- the proxy listens on the master service IP address and it sends traffic to the correct API server.
+  - creates a dummy network interface and assigns the `10.1.2.3` to it.
+  - removes connection tracking (conntrack) if iptables/nftables is enabled as the IP address is local to the `Node`.
+- the proxy listens on the `10.1.2.3` and using the [PROXY protocol](http://www.haproxy.org/download/2.0/doc/proxy-protocol.txt) it sends the data stream to the Envoy ingress gateway (EIGW).
+- EIGW listens for PROXY protocol on a dedicated `8443` port. EIGW reads the destination IP + port from the PROXY protocol and forwards traffic to the correct upstream apiserver.
 
 The sidecar is a standalone component. It's possible to transparently change the proxy implementation without any modifications to the sidecar. The simplified flow looks like:
 
 ```text
-                                              +------------------+
-                                              |                  |
-                                              | Shoot API Server |
-                                              |                  |
-                                              | Cluster A        |
-                                              |                  |
-                                              +------------------+
-                                                        ^
-                                                        |
-                                                        |
-                                                        |
-                                                        |
-  +-----------------------------------------------------------------------+
-                                                        |   Shoot Cluster
-                                                        |
-                                                        |
-                                                        |
-                                                        |
-                                                        |
-                                                        |
-   +---------------------+                   +---------------------+
-   |                     |                   |                     |
-   |  Pod talking to     |                   |                     |
-   |  the kubernetes     |                   |       Proxy         |
-   |  service            +------------------>+  No TLS termination |
-   |                     |                   |                     |
-   |                     |                   |                     |
-   +---------------------+                   +---------------------+
++------------------+                    +----------------+
+| Shoot API Server |       TCP          |   Envoy IGW    |
+|                  +<-------------------+ PROXY listener |
+| Cluster A        |                    |     :8443      |
++------------------+                    +-+--------------+
```

Comment:
IIUC the Envoy IGW PROXY listener is exposed by another Network LB from the cloud provider. So each seed cluster would have 2 Network LB for all the shoots. Is it correct?

Reply:
Each seed would only need `1` Network LB - the LoadBalancer would listen on `443` and `8443` and forward the traffic to `Envoy IGW`.

Additional network LB(s) might be need if we reach some limit on the cloud provider (max amount of TCP connections) or just for redundancy. In this case, the LBs would again point to the same `Envoy IGW`